x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_type_form.tsx

```diff
@@ -264,36 +271,39 @@ export const ActionTypeForm = ({
     }
   }, [minimumActionThrottle, minimumActionThrottleUnit, actionThrottle, actionThrottleUnit]);
 
-  useEffect(() => {
-    (async () => {
-      setAvailableActionVariables(
-        messageVariables
-          ? getAvailableActionVariables(
-              messageVariables,
-              summaryMessageVariables,
-              selectedActionGroup,
-              isSummaryAction
-            )
-          : []
-      );
-
-      const defaultParams = await getDefaultParams();
-      if (defaultParams) {
-        for (const [key, paramValue] of Object.entries(defaultParams)) {
-          const defaultAADParams: typeof defaultParams = {};
-          if (actionItem.params[key] === undefined || actionItem.params[key] === null) {
-            setActionParamsProperty(key, paramValue, index);
-            // Add default param to AAD defaults only if it does not contain any template code
-            if (typeof paramValue !== 'string' || !paramValue.match(/{{.*?}}/g)) {
-              defaultAADParams[key] = paramValue;
+  useEffect(
+    () => {
+      (async () => {
+        setAvailableActionVariables(
+          messageVariables
+            ? getAvailableActionVariables(
+                messageVariables,
+                summaryMessageVariables,
+                selectedActionGroup,
+                isSummaryAction
+              )
+            : []
+        );
+
+        const defaultParams = await getDefaultParams();
+        if (defaultParams) {
+          for (const [key, paramValue] of Object.entries(defaultParams)) {
+            const defaultAADParams: typeof defaultParams = {};
+            if (actionItem.params[key] === undefined || actionItem.params[key] === null) {
+              setActionParamsProperty(key, paramValue, index);
+              // Add default param to AAD defaults only if it does not contain any template code
+              if (typeof paramValue !== 'string' || !paramValue.match(/{{.*?}}/g)) {
+                defaultAADParams[key] = paramValue;
+              }
             }
+            setStoredActionParamsForAadToggle(defaultAADParams);
           }
-          setStoredActionParamsForAadToggle(defaultAADParams);
         }
-      }
-    })();
+      })();
+    },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [actionItem.group, actionItem.frequency?.summary]);
+    !isSystemAction ? [actionItem.group, actionItem.frequency?.summary] : []
```

Comment:
This feels dangerous. I think is better to but the check inside the `useEffect` and skip calling `setAvailableActionVariables` if it is a system action.